From fc87098b8a26b70e7a8cf7a0c56d7ccdc60f730d Mon Sep 17 00:00:00 2001
From: Simon Gerhards <Simon.Gerhards@gmail.com>
Date: Sat, 15 Mar 2014 19:43:15 +0100
Subject: [PATCH] Read at most 256 bit from /dev/random. Trying to read too
 much could slow down the startup. It is hashed to 256 bit anyway. See Bug
 #1169.

Signed-off-by: ronys <ronys@users.sourceforge.net>
---
 src/os/linux/rand.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/os/linux/rand.cpp b/src/os/linux/rand.cpp
index 9ea7a6a..4f0b945 100644
--- a/src/os/linux/rand.cpp
+++ b/src/os/linux/rand.cpp
@@ -74,7 +74,7 @@ void pws_os::GetRandomSeed(void *p, unsigned &slen)
     if (ent_avail) {
       unsigned ent_bits;
       if (ent_avail >> ent_bits && ent_bits >= 32) {
-        slen = ent_bits/8;
+        slen = ent_bits >= 256 ? 256/8 : ent_bits/8;
         data = new char[slen];
         ifstream rnd("/dev/random");
         if (rnd.read(data, slen))
-- 
1.9.2

